src/ts/main/src/install-drag-drop-action-adapters.ts

```diff
@@ -0,0 +1,37 @@
+import {storageAction} from "app/storage"
+import {Dispatch} from "redux"
+
+export function installDragDropActionAdapaters(document: HTMLDocument,
+                                               dispatch: Dispatch<storageAction.ReceiveDroppedFile>) {
+    document.body.ondragover = (e) => {
+        e.preventDefault()
+        e.stopPropagation()
+    }
+
+    document.body.ondrop = (e) => {
+        e.preventDefault()
+        e.stopPropagation()
+
+        //There's a lot that needs to be done to:
+        // - handle only file types we actually support
+        // - handle multiple files
+        // - put in place other defensiveness like checking files sizes and such
+        //
+        // This functionality should be considered prototype/demonstration-quality for now.
+
+        const fileList = e.dataTransfer!.files
+        for (let i = 0; i < fileList.length; i++) {
+            const reader = new FileReader()
+            reader.addEventListener('load', (event) => {
+                dispatch({
```

Comment:
do you need to unsubscribe the eventListener in here?

Reply:
Two things:

a) This code is very "practical", i.e. kind of a hack/PoC. So there's that.
b) This code should run exactly once, on single-page-app "boot". So I don't think there's any need for an unsubscribe, if that holds true.